packages/mdc-top-app-bar/index.js

```diff
@@ -70,33 +71,41 @@ class MDCTopAppBar extends MDCComponent {
    * @return {!MDCTopAppBarFoundation}
    */
   getDefaultFoundation() {
-    return new MDCTopAppBarFoundation(
-      /** @type {!MDCTopAppBarAdapter} */ (Object.assign({
-        hasClass: (className) => this.root_.classList.contains(className),
-        addClass: (className) => this.root_.classList.add(className),
-        removeClass: (className) => this.root_.classList.remove(className),
-        registerNavigationIconInteractionHandler: (evtType, handler) => {
-          if (this.navIcon_) {
-            this.navIcon_.addEventListener(evtType, handler);
-          }
-        },
-        deregisterNavigationIconInteractionHandler: (evtType, handler) => {
-          if (this.navIcon_) {
-            this.navIcon_.removeEventListener(evtType, handler);
-          }
-        },
-        notifyNavigationIconClicked: () => {
-          this.emit(strings.NAVIGATION_EVENT, {});
-        },
-        registerScrollHandler: (handler) => window.addEventListener('scroll', handler, util.applyPassive()),
-        deregisterScrollHandler: (handler) => window.removeEventListener('scroll', handler),
-        getViewportScrollY: () => window.pageYOffset,
-        getTotalActionItems: () =>
-          this.root_.querySelectorAll(strings.ACTION_ITEM_SELECTOR).length,
-      })
-      )
+    /** @type {!MDCTopAppBarAdapter} */
+    const adapter = /** @type {!MDCTopAppBarAdapter} */ (Object.assign({
+      hasClass: (className) => this.root_.classList.contains(className),
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      registerNavigationIconInteractionHandler: (evtType, handler) => {
+        if (this.navIcon_) {
+          this.navIcon_.addEventListener(evtType, handler);
+        }
+      },
+      deregisterNavigationIconInteractionHandler: (evtType, handler) => {
+        if (this.navIcon_) {
+          this.navIcon_.removeEventListener(evtType, handler);
+        }
+      },
+      notifyNavigationIconClicked: () => {
+        this.emit(strings.NAVIGATION_EVENT, {});
+      },
+      registerScrollHandler: (handler) => window.addEventListener('scroll', handler, util.applyPassive()),
+      deregisterScrollHandler: (handler) => window.removeEventListener('scroll', handler),
+      getViewportScrollY: () => window.pageYOffset,
+      getTotalActionItems: () =>
+        this.root_.querySelectorAll(strings.ACTION_ITEM_SELECTOR).length,
+    })
     );
+
+    let foundation;
+    if (this.root_.classList.contains(cssClasses.SHORT_CLASS)) {
```

Comment:
Have you run this by Lynn? Given our desire to avoid putting business logic on the component level, this seems to go against that, but I suppose having different components for each type would be overkill...

Reply:
We talked a bit about it over the past couple days. She asked me to create this PR so she could go over how I broke it up and start the discussion. I'm not sure where this block of code would really belong we don't want it in the Component.

Reply:
Actually I think this might be exactly the kind of logic we want to change per web platform

- Vanilla JavaScript: Read the DOM to decide which type of foundation to use
- React: Read a React prop to decide which type of foundation to use
- Internal Google JavaScript: Read some internal property available at initiate-time